### PR TITLE
[FIX] mrp: bom and uom

### PR DESCRIPTION
**Pull request by sle-odoo**
It was not possible to select another uom of the category since rev[0].
This is because if the variants are not enabled, `product_id` is always
empty.

We also took the opportunity of removing some useless boilerplate code
now that the domain is enforced in xml.

Closes #48687

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by sswapnesh** [comment]
Would be better if you don't remove method from Stable Version.
(As Uom will be always based on Product, So IMO we should remove default method in master)

File: addons/mrp/models/mrp_bom.py [hunk 1]
```diff
@@ -264,10 +248,9 @@ def _get_default_product_uom_id(self):
         digits='Product Unit of Measure', required=True)
     product_uom_id = fields.Many2one(
         'uom.uom', 'Product Unit of Measure',
-        default=_get_default_product_uom_id,
         required=True,
         help="Unit of Measure (Unit of Measure) is the unit of measurement for the inventory control", domain="[('category_id', '=', product_uom_category_id)]")
-    product_uom_category_id = fields.Many2one(related='product_id.uom_id.category_id')
+    product_uom_category_id = fields.Many2one(related='product_tmpl_id.uom_id.category_id')
```
Comment (sswapnesh) [inline addons/mrp/models/mrp_bom.py:253]: This is not needed as `product_tmpl_id` is already related to  `product_id.product_tmpl_id`
and also adding `product_tmpl_id` here will be slow as well.

**Comment by lathama** [comment]
@sle-odoo knowing that 13 is no longer updated, is this PR still needed?

**Comment by C3POdoo** [comment]
Dear @sle-odoo,

Thank you for your contribution but the version 13.0 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**